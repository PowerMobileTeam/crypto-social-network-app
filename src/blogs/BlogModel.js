import { MINDS_ASSETS_CDN_URI } from '../config/Config';
import api from '../common/services/api.service';
import BaseModel from '../common/BaseModel';
import UserModel from '../channel/UserModel';

import { LICENSES } from '../common/services/list-options.service';
/**
 * User model
 */
export default class BlogModel extends BaseModel {

  /**
   * observables
   */
  static observables = [
    'thumbs:down:count',
    'thumbs:up:count',
  ]

  /**
   * shallow observables
   */
  static observablesShallow = [
    'thumbs:down:user_guids',
    'thumbs:up:user_guids',
  ]

  /**
   * Get banner source
   * @param {string} size
   */
  getBannerSource(size='medium') {
    const uri = (this.thumbnail_src && this.thumbnail_src != 'https://cdn.minds.com/thumbProxy?src=&c=2708')?
      this.thumbnail_src:
      `${MINDS_ASSETS_CDN_URI}front/dist/assets/logos/placeholder-bulb.jpg`;

    return { uri, headers: api.buildHeaders()};
  }

  /**
   * Child models
   */
  childModels() {
    return {
      ownerObj: UserModel
    }
  }

  /**
   * Get the license text
   */
  getLicenseText() {
    const lic =  LICENSES.find(license => this.license == license.value);
    if (!lic) return;
    return lic.text;
  }
}