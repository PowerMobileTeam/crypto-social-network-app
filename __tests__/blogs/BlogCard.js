import 'react-native';
import React from 'react';
import { Platform, Linking } from 'react-native';
import { shallow } from 'enzyme';
import BlogCard from '../../src/blogs/BlogCard';
import blogFakeFactory from '../../__mocks__/fake/blogs/BlogFactory'
import BlogModel from '../../src/blogs/BlogModel';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

Linking.openURL = jest.fn();

/**
 * Tests
 */
describe('blog card component', () => {

  it('should renders correctly', () => {

    const blogEntity = BlogModel.create(blogFakeFactory(1));


    const blog = renderer.create(
      <BlogCard entity={blogEntity} />
    ).toJSON();

    expect(blog).toMatchSnapshot();
  });

  it('should nav to blog on ios', async (done) => {

    const blogEntity = BlogModel.create(blogFakeFactory(1));

    const navigation = {navigate: jest.fn()};

    try {
      const wrapper = shallow(
        <BlogCard entity={blogEntity} navigation={navigation}/>
      );

      Platform.OS = 'ios';

      // call method
      wrapper.instance().navToBlog();

      // expect fn to be called once
      expect(navigation.navigate).toBeCalledWith('BlogView', {blog:blogEntity});
      done();
    } catch(e) {
      done.fail(e);
    }
  });

  it('should open browser to blog on android', async (done) => {

    const blogEntity = BlogModel.create(blogFakeFactory(1));

    const navigation = {navigate: jest.fn()};

    try {
      const wrapper = shallow(
        <BlogCard entity={blogEntity} navigation={navigation}/>
      );

      Platform.OS = 'android';

      // call method
      wrapper.instance().navToBlog();

      // expect fn to be called once
      expect(Linking.openURL).toBeCalled();
      done();
    } catch(e) {
      done.fail(e);
    }
  });
});