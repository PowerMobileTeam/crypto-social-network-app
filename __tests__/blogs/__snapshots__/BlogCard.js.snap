// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog card component should renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "flexDirection": "column",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "flexDirection": "row",
          "height": 150,
          "width": "100%",
        },
        Object {
          "overflow": "hidden",
        },
      ]
    }
  >
    <FastImageView
      onFastImageError={undefined}
      onFastImageLoad={undefined}
      onFastImageLoadEnd={undefined}
      onFastImageLoadStart={undefined}
      onFastImageProgress={undefined}
      resizeMode="cover"
      source={
        Object {
          "headers": Object {
            "Cache-Control": "no-cache, no-store, must-revalidate",
            "Pragma": "no-cache",
          },
          "uri": "https://www.minds.com/fs/v1/banners/836607056186159104/1524838665",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flexDirection": "column",
        "padding": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#444",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "fontWeight": "800",
              "letterSpacing": 0.5,
            }
          }
        >
          Some Title
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 4,
              "paddingTop": 4,
            }
          }
        >
          <View
            activeOpacity={undefined}
            height={24}
            onLongPress={undefined}
            onPress={undefined}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "height": 24,
                  "paddingBottom": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": 24,
                },
                Object {
                  "borderRadius": 12,
                },
                undefined,
              ]
            }
            width={24}
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "backgroundColor": "rgba(0,0,0,0.2)",
                    "bottom": 0,
                    "flex": 1,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 12,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#888",
                "fontSize": 10,
                "paddingLeft": 8,
              }
            }
          >
            SOMEUSERMANE
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 8,
                "paddingLeft": 5,
              }
            }
          >
            Apr 27, 2018, 07:17
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
